src/cache.c

```diff
@@ -12,6 +12,18 @@ struct cache_entry *alloc_entry(char *path, char *content_type, void *content, i
     ///////////////////
     // IMPLEMENT ME! //
     ///////////////////
+
+    struct cache_entry *ce = malloc(sizeof *ce);
+
+    ce -> path = strdup(path);
+    ce -> content_type = strdup(content_type);
+    ce -> content = malloc(content_length);
```

Comment:
remember to make space for the null terminator in strings
`ce -> content = malloc(content_length + 1);`